pkg/proxy/backend/authenticator.go

```diff
@@ -210,12 +210,19 @@ func (auth *Authenticator) handshakeFirstTime(logger *zap.Logger, cctx ConnConte
 
 	// forward other packets
 	pluginName := ""
+	pktIdx := 0
 loop:
 	for {
 		serverPkt, err := forwardMsg(backendIO, clientIO)
 		if err != nil {
+			// tiproxy pp enabled, tidb pp disabled, tls disabled => invalid sequence
+			// tiproxy pp disabled, tidb pp enabled, tls disabled => invalid sequence
+			if pktIdx == 0 && errors.Is(err, pnet.ErrInvalidSequence) {
```

Comment:
Add `&& cfg.proxyProtocol` in these judgements?

Reply:
No, whether proxy-protocol is enabled or not, these errors occur and the errors are the same.
Say, tiproxy enable + tidb disable, or tiproxy disable + tidb enable, the error still occurs.

Reply:
But it is fixed now. We don't know any new issues triggering the error.

Reply:
The error message is a guess. It doesn't mean the reason must be it. Or we can overwrite the error message.

Reply:
How about "TiProxy fails to connect to TiDB, please make sure TiDB proxy-protocol is set correctly first. If this error still exists, please contact PingCAP"?